Xamarin.Essentials/Calendars/Calendars.uwp.cs

```diff
@@ -522,5 +522,67 @@ static async Task<bool> PlatformRemoveAttendeeFromEvent(CalendarEventAttendee ne
 
             return attendeeToRemove != null;
         }
+
+        static async Task<bool> PlatformAddReminderToEvent(CalendarEventReminder calendarEventReminder, string eventId)
+        {
+            await Permissions.RequestAsync<Permissions.CalendarWrite>();
+
+            var instance = await CalendarRequest.GetInstanceAsync();
+
+            Appointment uwpAppointment;
+            try
+            {
+                uwpAppointment = await instance.GetAppointmentAsync(eventId);
+            }
+            catch (ArgumentException)
+            {
+                if (string.IsNullOrWhiteSpace(eventId))
+                {
+                    throw new ArgumentException($"[UWP]: No Event found for event Id {eventId}");
+                }
+                else
+                {
+                    throw new ArgumentOutOfRangeException($"[UWP]: No Event found for event Id {eventId}");
+                }
+            }
+
+            uwpAppointment.Reminder = TimeSpan.FromMinutes(calendarEventReminder.MinutesPriorToEventStart);
+            var calendar = await instance.GetAppointmentCalendarAsync(uwpAppointment.CalendarId);
+            await calendar.SaveAppointmentAsync(uwpAppointment);
+            uwpAppointment = await instance.GetAppointmentAsync(eventId);
```

Comment:
Don't feel this is necessary - should wrap SaveAppointmentAsync in try catch and return false from catch if can't save

Reply:
After looking at how I was handling CalendarEvents, I believe when you 'save' the uwpAppointment it will actually update the event you pass in, so it might just be worth checking that uwpAppointment.Reminder != null as the return. (without the reget)

see:
await calendar.SaveAppointmentAsync(appointment);
if (!string.IsNullOrEmpty(appointment.LocalId))
   return appointment.LocalId;